File: frontend/types/patient.ts
import { Vitals, VitalSnapshot } from './vitals';

export type Gender = 'male' | 'female' | 'other';

export type PatientStatus = 'WAITING' | 'IN_REVIEW' | 'IN_TREATMENT' | 'COMPLETED';

export interface Patient {
  id: string;
  name: string;
  age: number;
  gender: Gender;
  chiefComplaint: string;
  symptoms: string[];
  medicalHistory: string[] | null;
  historyAvailable: boolean;
  arrivalTime: string;
  status: PatientStatus;
  createdAt: string;
  updatedAt: string;
  latestAssessment?: Assessment | null;
  latestVitals?: Vitals | null;
  vitalHistory?: VitalSnapshot[];
  assessmentHistory?: AssessmentSummary[];
}

export interface Assessment {
  id: string;
  priority: Priority;
  riskScore: number;
  confidence: number;
  safetyFlags: string[];
  reasons: ReasonItem[];
  recommendedAction: RecommendedAction;
  deteriorating: boolean;
  ageGroup: AgeGroup;
  createdAt: string;
  safetyFloor?: Priority | null;
  deteriorationSeverity?: string | null;
}

export interface AssessmentSummary {
  id: string;
  priority: Priority;
  riskScore: number;
  confidence: number;
  deteriorating: boolean;
  createdAt: string;
}

export type Priority = 'CRITICAL' | 'HIGH' | 'MODERATE' | 'LOW';
export type AgeGroup = 'PEDIATRIC' | 'ADULT' | 'GERIATRIC';
export type RecommendedAction =
  | 'IMMEDIATE_CLINICIAN_REASSESSMENT'
  | 'URGENT_CLINICIAN_REVIEW'
  | 'CLINICIAN_REVIEW'
  | 'ROUTINE_REVIEW';

export interface ReasonItem {
  code: string;
  message: string;
}

File: backend/app/schemas/patient.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from app.utils.enums import PatientStatus

class PatientCreate(BaseModel):
    name: str
    age: int = Field(..., ge=0, le=150)
    gender: str
    chiefComplaint: str
    symptoms: List[str] = []
    medicalHistory: Optional[List[str]] = None
    historyAvailable: bool = True

class PatientResponse(BaseModel):
    id: str
    name: str
    age: int
    gender: str
    chiefComplaint: str
    symptoms: List[str]
    medicalHistory: Optional[List[str]]
    historyAvailable: bool
    arrivalTime: datetime
    status: PatientStatus
    createdAt: datetime
    updatedAt: datetime

    model_config = {"from_attributes": True}

class PatientListItem(BaseModel):
    id: str
    name: str
    age: int
    gender: str
    chiefComplaint: str
    status: PatientStatus
    arrivalTime: datetime

    model_config = {"from_attributes": True}
